solr/core/src/test/org/apache/solr/response/transform/TestChildDocTransformer.java

```diff
@@ -242,10 +242,10 @@ private void testChildDocNonStoredDVFields() throws Exception {
         "fl", "*,[child parentFilter=\"subject:parentDocument\"]"), test1);
 
     assertJQ(req("q", "*:*", "fq", "subject:\"parentDocument\" ",
-        "fl", "subject,[child parentFilter=\"subject:parentDocument\" childFilter=\"title:foo\"]"), test2);
+        "fl", "id,_childDocuments_,subject,intDvoDefault,[child parentFilter=\"subject:parentDocument\" childFilter=\"title:foo\"]"), test2);
```

Comment:
Another major difference is that now since child documents are stored as fields, the user needs to explicitly add them to the list of return fields. I have no opinion regarding this issue, but this might pose a problem to some use cases. Perhaps documenting this would be enough?

Reply:
ah; interesting.  It's logical.  Is this only needed for anonymous child docs (thus \_childDocuments\_ or any/all possible relationship names that aren't necessarily just at the root level but anywhere in the hierarchy?  Perhaps this is where that "anonChildDocs" ought to come into play again for backwards-compatibility sake?  Well perhaps not... someone who is using anonymous child docs today will not have the nested field metadata and thus the old logic will kick in and ensure child documents are added as it was; right?

Reply:
This is what I would expect would happen. I am going to try it to check this works as expected. Would not want to suddenly change this for everyone :worried:

Reply:
I had to add a small addition and indeed, the behaviour was as expected.
I added a test to ensure there is backwards compatibility with the old XML format [here](https://github.com/apache/lucene-solr/blob/2573b89e465de615623084145cab7d17e9cb8a07/solr/core/src/test/org/apache/solr/response/transform/TestChildDocTransformer.java#L100).

Reply:
I was thinking about what you said, and perhaps we could add the old format using if the user did not index the field meta-data. I prefer having a flag, since the user can query separate two collections and get the same output if desired, instead of having to deal with two different XML formats.

Reply:
I think either the user hasn't yet started using the new key'ed/labeled style of child documents, or they have updated completely.  It's a migration to a new way you either do or don't do (and perhaps one day will not have a choice).

Reply:
Of course, one day this will probably get removed. Right now there is a way to provide the old XML format.

Reply:
I'm really not a fan of {{anonChildDocs}} flag; I regret I conjured up the idea.  If we have "nest" schema fields, the user wants nested documents (including field/label association), if the schema doesn't it ought to work as it used to.  I think this is straight-forward to reason about and document.